File: src/ruby/hello_world/hello_world.rb
# frozen_string_literal: true

require 'optparse'
require_relative 'lib/hello'

Options = Struct.new(:name)

args = Options.new("World")

OptionParser.new do |opts|
  opts.banner = "Usage: hello.rb [options]"
  opts.on("--name=NAME", "Name to greet") do |v|
    args.name = v
  end
end.parse!

puts(say_hello(args.name))

File: src/js/hello_world/hello_world.js
const { Command } = require("commander");

function greet(name) {
    return "Hello, " + name + "!";
}

function main() {
    const program = new Command();
    program.option("-n, --name <string>", "", "World");
    program.parse();
    const options = program.opts();
    console.log(greet(options.name));
}

main();
